### PR TITLE
Feat improve diff report

### PR DESCRIPTION
**Pull request by freshavocado7**
This PR aims to improve some of the information provided in the diff report.

- Display the parents of objects in order to provide more information.
- Provide `start, finish` and `source, target` information for objects, if applicable.
- Highlight deleted and new text in the description diff between the previous and current versions of the text accordingly.

### CODE REVIEW

File: capella_diff_tools/report.py [hunk 1]
```diff
@@ -107,6 +114,29 @@ def _traverse_and_diff(data):
...
+            text = _wrap_text(text, "removed")
+            previous_result += f"<removed>{text}</removed>"
+        elif operation == 1:
+            text = _wrap_text(text, "added")
+            current_result += f"<added>{text}</added>"
+    return previous_result, current_result
+
+
+def _wrap_text(text, tag):
+    return re.sub(r"(<[^>]+>)(.*?)(</[^>]+>)", rf"\1<{tag}>\2</{tag}>\3", text)
```
Comment (Wuestengecko) [inline capella_diff_tools/report.py]: This function is confusing me, what exactly is its purpose? It seems to just add the same `<added>` / `<removed>` tags a second time if the hunk looks like HTML ... but why?

Comment (Wuestengecko) [inline capella_diff_tools/report.py]: [Oi mate! Ya got a loicense fer dat?](https://regexlicensing.org/)

tl;dr: Don't mix HTML and regex.

```python
from capella_diff_tools import report

old = ['<p>Hello<span title="lol :>">;</span> World!</p>']
new = ["<p>Hello, World!</p>"]
result = report._diff_description(old, new)
print(*result, sep="\n\n")
```

```html
<p>Hello<removed><span title="lol :><removed>">;</removed></span></removed> World!</p>

<p>Hello<added>,</added> World!</p>
```

File: capella_diff_tools/report.py [hunk 2]
```diff
@@ -107,6 +114,29 @@ def _traverse_and_diff(data):
...
+    for operation, text in diff:
+        if operation == 0:
+            previous_result += text
+            current_result += text
+        elif operation == -1:
+            text = _wrap_text(text, "removed")
+            previous_result += f"<removed>{text}</removed>"
+        elif operation == 1:
+            text = _wrap_text(text, "added")
+            current_result += f"<added>{text}</added>"
```
Comment (Wuestengecko) [inline capella_diff_tools/report.py]: Here we're "inventing" new HTML elements, instead of using the standardized ins/del elements like elsewhere. The only difference is the styling we apply to it (ins/del gets color, added/removed gets background-color). I think we should unify this a bit, or at least use something other than the tag name to distinguish between foreground/background color (e.g. a CSS class).

File: capella_diff_tools/report.html.jinja [hunk 3]
```diff
@@ -49,6 +49,12 @@
                 border: 1px solid #ccc;
                 padding: 0.25em;
             }
+            removed{
```
Comment (Wuestengecko) [inline capella_diff_tools/report.html.jinja]: Inconsistent formatting.

```diff
-            removed{
+            removed {
```

File: capella_diff_tools/report.py (hunk 1 above)
Comment (freshavocado7) [reply capella_diff_tools/report.py]: Sometimes there is the case that a whole `<p>...</p>` paragraph is inside a `<removed>` or `<ins>` tag making the styling obsolete. This function was rewrapping the inside of the html if there were opening and closing tags in it. This also applied to the `del` and ìns` tag, where the background color was not applying.

Comment (freshavocado7) [reply capella_diff_tools/report.py]: So when is it appropriate using a regex? based on this site even when owning a license its good to avoid it.

**Review by Wuestengecko — COMMENTED** [review]
LGTM.
